handlers.go

```diff
@@ -11,11 +11,14 @@ import (
 	"io/ioutil"
 	"log"
 	"net/http"
+	"sync"
 )
 
 // A autographer signs input data with a private key
 type autographer struct {
-	signers []Signer
+	signers    []Signer
+	nonces     []nonce
+	noncesLock sync.Mutex
```

Comment:
Mostly because I think we'll need more than one lock to do various operations without locking the entire handler. I don't want to run in a situation where we lock the handler regularly for things that don't necessarily impact normal operation.